src/trie.js

```diff
@@ -100,8 +100,10 @@ var marexandre;
             // Check if next character exists in children, and if does dive deeper
             if (copy[j + 1]) {
               var exists2 = tmpTrie.children.hasOwnProperty(copy[j + 1].toString());
-              if (tmpTrie.is_end && !exists2) {
+              if (tmpTrie.is_end) {
```

Comment:
Could you rename `tmpTrie`, to `currentNode` ? I think it makes more sense and is clearer.
Also, I would rename `copy` to `remainingText`.
Maybe add a method `hasChildrenWithValue(char)` to whatever type `tmpTrie` is, to clarify and remove duplication.

Don't want to make too many changes, but it's good to clarify a bit the code for the next time we have to change it.